src/tests/collate_tests/test_collate.py

```diff
@@ -4,20 +4,32 @@
 Unit tests for `collate` module.
 
 """
-from triage.component.collate import Aggregate, Aggregation
-
+from triage.component.collate import Aggregate, Aggregation, Categorical
 
 def test_aggregate():
     agg = Aggregate("*", "count", {})
     assert list(map(str, agg.get_columns())) == ["count(*)"]
 
+def test_aggregate_cast():
+    agg = Aggregate("*", "count", {}, coltype="REAL")
+    assert list(map(str, agg.get_columns())) == ["count(*)::REAL"]
+
+def test_categorical_cast():
+    cat = Categorical("c", ['A','B','C'], "sum", {}, coltype="SMALLINT")
+    assert list(map(str, cat.get_columns())) == ["sum((c = 'A')::INT)::SMALLINT","sum((c = 'B')::INT)::SMALLINT","sum((c = 'C')::INT)::SMALLINT"]
```

Comment:
emacs doesn't format your lines huh? :stuck_out_tongue: 

I'm sure the right operand can put its list elements on new lines in order to make this nicer :wink: